### ISSUE TITLE
Question about the learnable image tensor of in-context tuning

### ISSUE THREAD

**Issue by YangHan-Morningstar** [baaivision/Painter#18]
Hi there, thanks for your amazing work. After reading your paper of SegGPT. I'm little confused about the in-context tuning. In the paper, during the training stage, SegGPT treat a learnable image tensor as learnable prompt. But in the normal training stage, the input is a pair of in-context images with each mask, such as image1-mask1 and image2-mask2. So the learnable image tensor is a random image-mask? With image3-mask3 from the datasets, the whole input is image-mask(prompt) and image3-mask3? Due to the mask of random image-mask is random, so there is no table for loss calculation and gradient backward, how does it be trained? Please tell me more and help me solve this. Thanks!

**Comment by SteveImmanuel** [comment]
This is [my implementation](https://github.com/SteveImmanuel/SegGPT-FineTune/blob/main/model.py) based on my understanding.

I think that once you used the learnable prompt, you simply replace the image1-mask1 with the image tensor that you optimize

**Comment by xizh01** [comment]
thanks for the amazing work. Only a small question regarding to in-context tuning. 

If image1-mask1, image2-mask2 present the input tensor of the model, which come from a new domain. image1-mask1, image2-mask2 refer to a pair of example image and query image, show in Fig. 2 (right to the text "Mix and Mask").

Question: In in-context tuning, image1 and mask1 **both** will be be optimized using the the ground truth of mask2?
